mne/decoding/time_gen.py

```diff
@@ -387,40 +388,88 @@ def _predict_time_loop(X, estimators, cv, slices, predict_mode):
     """
     n_epochs = len(X)
     # Loop across testing slices
-    y_pred = [list() for _ in range(len(slices))]
 
-    # XXX EHN: This loop should be parallelized in a similar way to fit()
+    if predict_mode == 'cross-validation':
+        # Check that training cv and predicting cv match
+        if (len(estimators) != len(cv)) or (cv.n != n_epochs):
+            raise ValueError(
+                'When `predict_mode = "cross-validation"`, the training '
+                'and predicting cv schemes must be identical.')
+        all_test = np.concatenate(list(zip(*cv))[-1])
+        test_slices = []
+        start = 0
+        for _, test in cv:
+            l = len(test)
+            stop = start + l
+            test_slices.append(slice(start, stop, 1))
+            start += l
```

Comment:
We should clarify the variables to better distinguish `test` epochs (from the cv) and `test` times